source/more_sampling_tools.Rmd

```diff
@@ -127,40 +126,53 @@ rnd = np.random.default_rng()
 
 ```{python}
 # The 3 buckets, each representing two trunks on a ship.
-bucket1 = ['Gold', 'Gold']  # Trunks in first ship.
-bucket2 = ['Gold',  'Silver']  # Trunks in second ship.
-bucket3 = ['Silver', 'Silver']  # Trunks in third ship.
+ship_chests = [
```

Comment:
I get that, but selecting variables in a for loop really sets a bad example.

Reply:
I think we have to accept that some of the code will be less than perfectly idiomatic in the early stages of the book - otherwise we're going to find ourselves cramming too much Python into the introductory chapters and losing an important part of our audience.

I was trying to think of a way of adding the idiomatic implementation somewhere else, for people to look at - like the idiomatic implementation of the Monty Hall, with a reference saying something like "# Actually, here it would be better form to store the ships as a list of lists, but this needs a little more explanation than we have space for here.   See X for a better implementation".

Reply:
I've put in a side-note - in the main PR - but I think I'll convert it to a foot-note.